h2o-py/h2o/frame.py

```diff
@@ -509,7 +509,10 @@ def logical_negation(self):
 
 
     def _unop(self, op):
-        return H2OFrame._expr(expr=ExprNode(op, self), cache=self._ex._cache)
+        ret = H2OFrame._expr(expr=ExprNode(op, self), cache=self._ex._cache)
+        if ret._ex._cache._names is not None:
```

Comment:
Intent is to ensure names of the original frame do not get carried over to a new one after a unop. For example, if I do the following:

```
    dataframe = {'A': [1,0,3,4], 'B': [5,6,-6, -1], 'C':[-4, -6, -7, 8]}
    frame = h2o.H2OFrame(dataframe)
    frame_asin = frame.asin()
```
I would expect the names of `frame_asin` to now be `asin(A), asin(B), and asin(C)`. However, before this change that was not the case.